core/src/main/scala/org/apache/spark/shuffle/sort/BypassMergeSortShuffleWriter.scala

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.shuffle.sort
+
+import java.io.{File, FileInputStream, FileOutputStream}
+
+import org.apache.spark._
+import org.apache.spark.executor.ShuffleWriteMetrics
+import org.apache.spark.serializer.Serializer
+import org.apache.spark.storage.{BlockId, BlockManager, BlockObjectWriter}
+import org.apache.spark.util.Utils
+import org.apache.spark.util.collection._
+
+/**
+ * This class handles sort-based shuffle's `bypassMergeSort` write path, which is used for shuffles
```

Comment:
I'll rewrite this Scaladoc to provide a better explanation and to also address how this is different than HashShuffleWriter (the key difference is the fact that this output in a format that can be served with IndexShuffleBlockResolver).